protocols/frost/keygen/round2.go

```diff
@@ -0,0 +1,97 @@
+package keygen
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/taurusgroup/cmp-ecdsa/pkg/math/curve"
+	"github.com/taurusgroup/cmp-ecdsa/pkg/math/polynomial"
+	"github.com/taurusgroup/cmp-ecdsa/pkg/message"
+	"github.com/taurusgroup/cmp-ecdsa/pkg/party"
+	"github.com/taurusgroup/cmp-ecdsa/pkg/round"
+	"github.com/taurusgroup/cmp-ecdsa/pkg/types"
+	zksch "github.com/taurusgroup/cmp-ecdsa/pkg/zk/sch"
+)
+
+// This round corresponds with steps 5 of Round 1, 1 of Round 2, Figure 1 in the Frost paper:
+//   https://eprint.iacr.org/2020/852.pdf
+type round2 struct {
+	*round1
+	// f_i is the polynomial this participant uses to share their contribution to
+	// the secret
+	f_i *polynomial.Polynomial
+	// Phi contains the polynomial commitment for each participant, ourselves included.
+	//
+	// Phi[l][k] corresponds to phi_lk in the Frost paper.
+	Phi map[party.ID]*polynomial.Exponent
+}
+
+// ProcessMessage implements round.Round.
+func (r *round2) ProcessMessage(l party.ID, content message.Content) error {
+	msg, ok := content.(*Keygen2)
+	if !ok {
+		return fmt.Errorf("failed to convert message to Keygen2: %v", msg)
+	}
```

Comment:
Technically this should not happen, but better safe than sorry 😉

Reply:
Well, the alternative would be to panic, if it's really a programmer error. This felt more apt :P